kin-backup-and-restore/kin-backup-and-restore-ui-tests/src/androidTest/kotlin/kin/backup/and/restore/ui/tests/BackupActivityValidateSwitchBetweenScreens.kt

```diff
@@ -0,0 +1,263 @@
+package kin.backup.and.restore.ui.tests
+
+import android.app.Activity
+import android.content.ComponentName
+import android.content.Intent
+import android.support.test.InstrumentationRegistry
+import android.support.test.InstrumentationRegistry.getInstrumentation
+import android.support.test.espresso.Espresso.closeSoftKeyboard
+import android.support.test.espresso.Espresso.onView
+import android.support.test.espresso.action.ViewActions.click
+import android.support.test.espresso.action.ViewActions.replaceText
+import android.support.test.espresso.assertion.ViewAssertions.doesNotExist
+import android.support.test.espresso.assertion.ViewAssertions.matches
+import android.support.test.espresso.intent.Intents
+import android.support.test.espresso.intent.Intents.intended
+import android.support.test.espresso.intent.matcher.IntentMatchers.hasComponent
+import android.support.test.espresso.matcher.ViewMatchers.*
+import android.support.test.filters.LargeTest
+import android.support.test.rule.ActivityTestRule
+import android.support.test.runner.AndroidJUnit4
+import android.support.test.uiautomator.UiDevice
+import android.widget.TextView
+import kin.backup.and.restore.ui.tests.UiTestUtils.childAtPosition
+import kin.backup.and.restore.ui.tests.UiTestUtils.hasValueEqualTo
+import kin.backup.and.restore.ui.tests.UiTestUtils.isSameDrawable
+import kin.backupandrestore.backup.view.BackupActivity
+import kin.sdk.Environment
+import kin.sdk.KinClient
+import org.hamcrest.Matchers.*
+import org.hamcrest.core.IsInstanceOf
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import kotlin.test.assertEquals
+
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
+class BackupActivityValidateSwitchBetweenScreens {
+
+    private lateinit var kinClient: KinClient
+    private lateinit var stubIntent: Intent
+    private var componentName: ComponentName? = null
+
+    @Rule
+    @JvmField
+    var activityTestRule = ActivityTestRule(BackupActivity::class.java, false, false)
+
+
+    @Before
+    fun setup() {
+        kinClient = KinClient(InstrumentationRegistry.getTargetContext(), Environment.TEST, UiTestUtils.appId, UiTestUtils.storeKey)
+        kinClient.clearAllAccounts()
+        val kinAccount = kinClient.addAccount()
+        stubIntent = UiTestUtils.createStubIntent(BackupActivity::class.java.name, kinAccount.publicAddress)
```

Comment:
so we're testing here backup activity without real data? just checking the password widget nad transitions between screens?
in words how does `KinClient` gets attached to the activity/backup manager?

Reply:
after looking in UiTestUtils I get it, we recreate `KinClient` using env params